eZ/Publish/SPI/Search/Indexing.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
+
+/**
+ * Interface for indexing in search backend.
+ */
+interface Indexing
+{
+    /**
+     * Purge search engine index.
+     */
+    public function purgeIndex();
+
+    /**
+     * Make indexing operation permanent if Search Handler needs and supports it.
+     * Note: it is a responsibility of a Search Handler to decide if this is needed and how to do this.
+     */
+    public function commitIndex();
```

Comment:
I would rather go with resuing `commit` but we can indeeed skip the flush argument:
```php
    /**
     * Commits changes to search index, making it available for search.
     *
     *  Search engines implementing this should expose configuration to specify if this should
     * do a flush, aka hard commit, meaning changes should be forced to be written to durable
     * storage or not. By default this should be the case, but if transaction logs are enabled
     * it can be safe to optimize performance and allow disabling flushing.
     * 
     * @param bool $forceFlush
     */
    public function commit();
```